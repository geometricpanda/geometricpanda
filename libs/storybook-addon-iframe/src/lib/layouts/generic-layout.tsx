import React, { ReactNode } from 'react';
import styled from 'styled-components';

const GenericIconLayoutWrapper = styled.div`
  width: 200px;
  margin-bottom: 1rem;
`;

const GenericLayoutText = styled.h1`
  color: #999;
  font-size: 1rem;
  font-weight: bold;
`;

const GenericLayoutContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  height: 100%;
  width: 100%;
`;

export interface GenericLayoutProps {
  icon?: ReactNode;
}

export const GenericLayout: React.FC<GenericLayoutProps> = ({ children, icon }) => (
  <GenericLayoutContainer>
    {icon && (
      <GenericIconLayoutWrapper>
        {icon}
      </GenericIconLayoutWrapper>
    )}
    <GenericLayoutText>
      {children}
    </GenericLayoutText>
  </GenericLayoutContainer>
);
