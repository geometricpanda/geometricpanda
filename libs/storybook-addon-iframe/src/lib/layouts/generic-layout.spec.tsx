import renderer from 'react-test-renderer';
import React from 'react';
import { GenericLayout } from './generic-layout';

describe('GenericLayout', () => {

  const TestIcon = () => <img src={null} alt='test-icon'/>;
  const TestText = () => <p>Lorem Ipsum</p>;

  it('should render and icon and text', () => {
    const layout = (
      <GenericLayout icon={<TestIcon />}>
        <TestText />
      </GenericLayout>
    );

    const tree = renderer.create(layout);
    const testInstance = tree.root;

    expect(testInstance.findByType(TestIcon)).toBeTruthy();
    expect(testInstance.findByType(TestText)).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

  it('should render text', () => {
    const layout = (
      <GenericLayout>
        <TestText />
      </GenericLayout>
    );

    const tree = renderer.create(layout);
    const testInstance = tree.root;

    expect(testInstance.findByType(TestText)).toBeTruthy();
    expect(tree).toMatchSnapshot();
  });

});
