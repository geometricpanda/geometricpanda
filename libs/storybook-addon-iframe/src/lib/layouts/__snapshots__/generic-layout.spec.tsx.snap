// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericLayout should render and icon and text 1`] = `
<div
  className="es7h93-2 fWpopI"
>
  <div
    className="es7h93-0 fOAtDc"
  >
    <img
      alt="test-icon"
      src={null}
    />
  </div>
  <h1
    className="es7h93-1 eKtWtv"
  >
    <p>
      Lorem Ipsum
    </p>
  </h1>
</div>
`;

exports[`GenericLayout should render text 1`] = `
<div
  className="es7h93-2 fWpopI"
>
  <h1
    className="es7h93-1 eKtWtv"
  >
    <p>
      Lorem Ipsum
    </p>
  </h1>
</div>
`;
